src/TrackProgress/SFA.DAS.TrackProgress/Application/Commands/TrackProgressCommand.cs

```diff
@@ -14,24 +14,47 @@ public record TrackProgressCommand(
     DateTime PlannedStartDate,
     ProgressDto? Progress) : IRequest<TrackProgressResponse>;
 
+public record ErrorDetail(
+    string error,
+    string description
+    );
+
 public class TrackProgressResponse
 {
     public HttpStatusCode StatusCode { get; set; }
     public string Message { get; set; } = string.Empty;
+    public string ErrorTitle { get; set; } = string.Empty;
+    public List<ErrorDetail> ErrorDetails { get; set; } = new List<ErrorDetail>();
 
     public IActionResult Result
         => StatusCode switch
         {
             HttpStatusCode.Created => new CreatedResult(string.Empty, null),
             HttpStatusCode.NotFound => new NotFoundObjectResult(Message),
+            HttpStatusCode.BadRequest => new BadRequestObjectResult(GetProblemDetails()),
```

Comment:
Couple of tiny comments, not directly about your code. NotFound doesn't need any message content. If you add it the consuming applications (depending on how they are written may have to handle it). 

The second point is the error raised by the commitments search is being exposed to the consumer (this would be viewed as a security hole). If the commitments search fails. It would probably be better to log the failure and return a 405

Reply:
Agree, changed to 500 internal server error in this instance